compiler/rustc_mir/src/transform/coverage/mod.rs

```diff
@@ -512,6 +508,8 @@ fn make_code_region(
     } else {
         source_file.lookup_file_pos(span.hi())
     };
+    let start_line = source_map.doctest_offset_line(&source_file.name, start_line);
```

Comment:
These look like the only pertinent changes in this file. Let me know if I'm missing something.

Very exciting if this works.